include/bsp/iotc_bsp_io_fs.h

```diff
@@ -32,223 +29,175 @@
+ * @param [in] open_flags A {@link ::iotc_bsp_io_fs_open_flags_t file operation}
+ *     bitmask.
+ * @param [out] resource_handle_out A
+ *     {@link ::iotc_bsp_io_fs_resource_handle_t handle to an open file}.
  */
 iotc_bsp_io_fs_state_t iotc_bsp_io_fs_open(
     const char* const resource_name, const size_t size,
     const iotc_bsp_io_fs_open_flags_t open_flags,
     iotc_bsp_io_fs_resource_handle_t* resource_handle_out);
 
 /**
- * @function
- * @brief Read a file.
- *
- * The Device SDK passes an iotc_bsp_io_fs_resource_handle_t handle to
- * this function in order to read a file. Before calling this function,
- * the Device SDK calls the <code>iotc_bsp_io_fs_open()</code> function
- * to create an iotc_bsp_io_fs_resource_handle_t handle.
+ * @brief Reads a file.
  *
- * Implementations of this function must fill the buffer parameter at
- * offset 0.
+ * @details The function must fill the buffer at offset 0. The function can
+ * allocate buffers by:
+ *     - Allocating the buffer once, resusing it at each function call,
+ *       and freeing it when the file closes.
+ *     - Creating a new buffer each time the function is called and
+ *       and freeing the old buffer before the function returns.
  *
- * Implementations of this function can allocate a buffer once, reuse the
- * buffer, and free the buffer when the file is closed. Or, implementations can
- * create a new buffer on repeated calls and free the previous one before
- * returning from this function.
- *
- * @param [in] resource_handle The iotc_bsp_io_fs_resource_handle_t handle from
- *     <code>iotc_bsp_io_fs_open()</code>.
+ * @param [in] resource_handle A
+ *     {@link ::iotc_bsp_io_fs_resource_handle_t handle to an open file}.
  * @param [in] offset The position within the resource, in bytes, from which
  *     to start read operations.
  * @param [out] buffer A pointer to a buffer with the bytes read from the file.
- *     The buffer is already allocated by the Device SDK.
+ *     Function implementations must define how to allocate buffers The buffer
+ *     is already allocated by the Device SDK.
  * @param [out] buffer_size The number of bytes read from the file and stored
  *     in the buffer.
- *
- * @see iotc_bsp_io_fs_open
- * @see iotc_bsp_io_fs_close
- * @see iotc_bsp_io_fs_state_e
- *
- * @return iotc_bsp_io_fs_state_t File management function status.
  */
 iotc_bsp_io_fs_state_t iotc_bsp_io_fs_read(
     const iotc_bsp_io_fs_resource_handle_t resource_handle, const size_t offset,
     const uint8_t** buffer, size_t* const buffer_size);
 
 /**
- * @function
- * @brief Write to a file.
+ * @details Writes to a file.
  *
- * Implementations of this function can allocate a buffer once, reuse the
- * buffer, and close the file to free the buffer. Or, implementations can
- * create a new buffer and free the previous one before returning from this
- * function.
+ * @details The function must fill the buffer at any offset. The function can
```

Comment:
This does not fit here. This belongs only to the read function. The write function simply does not allocate or fill any buffers. (I see this was here before, but the best action here is to remove. I think this was here by mistake.)
`[delete section start]`

Reply:
Done